platform/libodp-linux.map

```diff
@@ -0,0 +1,600 @@
+	odp_thrmask_from_str;
+	odp_thrmask_isset;
+	odp_thrmask_last;
+	odp_thrmask_next;
+	odp_thrmask_or;
+	odp_thrmask_set;
+	odp_thrmask_setall;
+	odp_thrmask_to_str;
+	odp_thrmask_worker;
+	odp_thrmask_xor;
+	odp_thrmask_zero;
+	odp_ticketlock_init;
+	odp_ticketlock_is_locked;
+	odp_ticketlock_lock;
+	odp_ticketlock_trylock;
+	odp_ticketlock_unlock;
+	odp_time_cmp;
+	odp_time_diff;
+	odp_time_diff_ns;
+	odp_time_global;
+	odp_time_global_from_ns;
+	odp_time_global_res;
+	odp_time_local;
+	odp_time_local_from_ns;
+	odp_time_local_res;
+	odp_timeout_alloc;
+	odp_timeout_free;
+	odp_timeout_fresh;
+	odp_timeout_from_event;
+	odp_timeout_tick;
+	odp_timeout_timer;
+	odp_timeout_to_event;
+	odp_timeout_to_u64;
+	odp_timeout_user_ptr;
+	odp_timer_alloc;
+	odp_timer_cancel;
+	odp_timer_capability;
+	odp_timer_current_tick;
+	odp_timer_free;
+	odp_timer_ns_to_tick;
+	odp_timer_pool_create;
+	odp_timer_pool_destroy;
+	odp_timer_pool_info;
+	odp_timer_pool_start;
+	odp_timer_pool_to_u64;
+	odp_timer_set_abs;
+	odp_timer_set_rel;
+	odp_timer_tick_to_ns;
+	odp_timer_to_u64;
+	odp_time_sum;
+	odp_time_to_ns;
+	odp_time_wait_ns;
+	odp_time_wait_until;
+	odp_tm_capabilities;
+	odp_tm_capability;
+	odp_tm_create;
+	odp_tm_destroy;
+	odp_tm_drop_prec_marking;
+	odp_tm_ecn_marking;
+	odp_tm_egress_init;
+	odp_tm_enq;
+	odp_tm_enq_with_cnt;
+	odp_tm_find;
+	odp_tm_is_idle;
+	odp_tm_node_connect;
+	odp_tm_node_context;
+	odp_tm_node_context_set;
+	odp_tm_node_create;
+	odp_tm_node_destroy;
+	odp_tm_node_disconnect;
+	odp_tm_node_fanin_info;
+	odp_tm_node_info;
+	odp_tm_node_lookup;
+	odp_tm_node_params_init;
+	odp_tm_node_sched_config;
+	odp_tm_node_shaper_config;
+	odp_tm_node_threshold_config;
+	odp_tm_node_wred_config;
+	odp_tm_priority_query;
+	odp_tm_priority_threshold_config;
+	odp_tm_queue_connect;
+	odp_tm_queue_context;
+	odp_tm_queue_context_set;
+	odp_tm_queue_create;
+	odp_tm_queue_destroy;
+	odp_tm_queue_disconnect;
+	odp_tm_queue_info;
+	odp_tm_queue_params_init;
+	odp_tm_queue_query;
+	odp_tm_queue_sched_config;
+	odp_tm_queue_shaper_config;
+	odp_tm_queue_threshold_config;
+	odp_tm_queue_wred_config;
+	odp_tm_requirements_init;
+	odp_tm_sched_create;
+	odp_tm_sched_destroy;
+	odp_tm_sched_lookup;
+	odp_tm_sched_params_init;
+	odp_tm_sched_params_read;
+	odp_tm_sched_params_update;
+	odp_tm_shaper_create;
+	odp_tm_shaper_destroy;
+	odp_tm_shaper_lookup;
+	odp_tm_shaper_params_init;
+	odp_tm_shaper_params_read;
+	odp_tm_shaper_params_update;
+	odp_tm_stats_print;
+	odp_tm_threshold_create;
+	odp_tm_threshold_destroy;
+	odp_tm_threshold_params_init;
+	odp_tm_thresholds_lookup;
+	odp_tm_thresholds_params_read;
+	odp_tm_thresholds_params_update;
+	odp_tm_total_query;
+	odp_tm_total_threshold_config;
+	odp_tm_vlan_marking;
+	odp_tm_wred_create;
+	odp_tm_wred_destroy;
+	odp_tm_wred_lookup;
+	odp_tm_wred_params_init;
+	odp_tm_wred_params_read;
+	odp_tm_wred_params_update;
+	odp_version_api_str;
+	odp_version_impl_name;
+	odp_version_impl_str;
```

Comment:
Must these be enumerated individually? The ODP API symbols should all match the pattern `odp_*`.

Reply:
Yes. However we must be sure to export correct symbols with correct versions.

Reply:
Not sure I follow. The concern is that maintaining this list would impose a burden as every new API defined (or removed) would have to remember to update this list.

Reply:
@Bill-Fischofer-Linaro is that really a "burden"? First commit does exactly that, however it makes impossible to stop library from exporting extra `_odp_*` symbols (which makes ABI definition loose).  Having such list would allow us to have proper symbol versioning, if we decide to have stable ABI (and we are almost there, judging from library ABI tracking).

Reply:
will Travis capture error if some function is missed in this list?

Reply:
It's a burden because it requires a separate list to be maintained apart from the main API definitions contained in the `.h` files, as well as the associated files lists we maintain in the various `Makefile.am` files.  Whenever you have to keep multiple lists in sync that's always an error-prone process unless these are auto-generated by scripts or other build tools.

The original idea behind the visibility controls was to make all symbols internal by default and only export symbols defined between the `visibility_begin.h` and `visibility_end.h` include files. It seems this has a similar goal, but via a different enumeration. It's not clear what we're gaining here vs. using a pattern since all ODP APIs share the `odp_` prefix.

Reply:
@muvarov no, as far as I remember. I will check, when I get back to this PR.
@Bill-Fischofer-Linaro this list should not be autogenerated, it binds symbols to library versions. Consider [libc](https://sourceware.org/git/?p=glibc.git;a=blob;f=math/Versions;h=01ed388ae5dc05115713dbb3c7abfa88b649e260;hb=HEAD), [GnuTLS](https://gitlab.com/gnutls/gnutls/blob/master/lib/libgnutls.map], [libgcrypt](https://git.gnupg.org/cgi-bin/gitweb.cgi?p=libgcrypt.git;a=blob;f=src/libgcrypt.vers;h=edf8a76ae72b9c74431c1b85567f9d33369a45c8;hb=HEAD).

In theory we can go [nettle](https://git.lysator.liu.se/nettle/nettle/blob/master/libnettle.map.in) way. However this limits usefullness of version mapping (versions are limited to the same id).

Reply:
@muvarov no, it does not capture an error. It can be implemented though.

@Bill-Fischofer-Linaro as I wrote, second patch is to be discussed.

Reply:
@lumag OK, we'll cover this on Monday's ARCH call.

Reply:
That is the same solution that DPDK has and we need such changes. But we should clear understand  how to maintain that and use. It's not part of API, it's  implementation linked map script. API will continue use visibility_begin/end.h so that implementation will not export it's functions. So we use it for linux-generic implementation.  Then should we use versions inside .map file? Also I as Bill not sure if we need list of all functions. Instead of export all odp_ and depricated_odp_ functions.

Reply:
We might use symbol versions, if we try to provide ABI backwards-compatibility between releases (we do not for now). Otherwise fist patch should be enough.